pkg/core/interop_neo.go

```diff
@@ -119,22 +119,22 @@ func contractUpdate(ic *interop.Context) error {
 	if contract == nil {
 		return errors.New("contract doesn't exist")
 	}
-	script := ic.VM.Estack().Pop().Bytes()
-	if len(script) > MaxContractScriptSize {
-		return errors.New("the script is too big")
+	script := ic.VM.Estack().Pop().BytesOrNil()
+	if l := len(script); script != nil && (l == 0 || l > MaxContractScriptSize) {
```

Comment:
Two separate checks would simplify that considerably.

Reply:
Fixed.